### ISSUE TITLE
How to open/close dialog programatically?

### ISSUE THREAD

**Issue by softmarshmallow** [radix-ui/primitives#785]
E.g. 

```tsx
openDialog(<MyDialog/>).then(d => {
    // do something with dialog result
})
```

**Comment by andy-hook** [comment]
Hi @softmarshmallow, If you need to manage this state yourself you can use the [controlled props](https://www.radix-ui.com/docs/primitives/components/dialog#root) on the `Root`.

```jsx
import * as Dialog from '@radix-ui/react-dialog';

...

const [open, setOpen] = React.useState(false);

return (
  <Dialog.Root open={open} onOpenChange={setOpen}>
    ...
  </Dialog.Root>
);
```

I'm not clear on your exact requirement just from your example but you should be able to achieve what you need with those, will close this for now but feel free to re-open if that's not what you're looking for.

**Comment by erickriva** [comment]
I understood what @softmarshmallow said... I was having the same problem a few days ago.
So I implemented this, maybe not the best solution but it worked for me.

It is about resolving a promise when clicking on a button inside Dialog/AlertDialog;

https://codesandbox.io/s/alert-dialog-promise-69kpqp?file=/src/MyComponent.tsx

**Comment by spbro007** [comment]
@andy-hook long live my brother till 59th century solving these bugs. spreading knowledge
tldr; It worked
